internal/controllers/ingress_controller_test.go

```diff
@@ -108,3 +108,90 @@ func TestIngressReconcilerIngressToEdge(t *testing.T) {
 		assert.ElementsMatch(t, testCase.edge.Routes, edge.Routes)
 	}
 }
+
+func TestIngressToTunnels(t *testing.T) {
+
+	testCases := []struct {
+		testName string
+		ingress  *netv1.Ingress
+		tunnels  []ingressv1alpha1.Tunnel
+	}{
+		{
+			testName: "Returns empty list when ingress is nil",
+			ingress:  nil,
+			tunnels:  []ingressv1alpha1.Tunnel{},
+		},
+		{
+			testName: "Returns empty list when ingress has no rules",
+			ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress",
+					Namespace: "test-namespace",
+				},
+				Spec: netv1.IngressSpec{
+					Rules: []netv1.IngressRule{},
+				},
+			},
+			tunnels: []ingressv1alpha1.Tunnel{},
+		},
+		{
+			ingress: &netv1.Ingress{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-ingress",
+					Namespace: "test-namespace",
+				},
+				Spec: netv1.IngressSpec{
+					Rules: []netv1.IngressRule{
+						{
+							Host: "my-test-tunnel.ngrok.io",
+							IngressRuleValue: netv1.IngressRuleValue{
+								HTTP: &netv1.HTTPIngressRuleValue{
+									Paths: []netv1.HTTPIngressPath{
+										{
+											Path: "/",
+											Backend: netv1.IngressBackend{
+												Service: &netv1.IngressServiceBackend{
+													Name: "test-service",
+													Port: netv1.ServiceBackendPort{
+														Number: 8080,
+													},
```

Comment:
It might be worth having the test case include multiple of the same service/port combination and make sure it only creates 1 and doesn't error since thats the one _interesting case_ i see

Reply:
Good catch, I'll add one of those tests

Reply:
Did some cleanup of the tests. Should now be covered by https://github.com/ngrok/ngrok-ingress-controller/pull/95/commits/5d2dca984d4636d3579ee29d345c2adde9053e50